Make tasks_per_node optional

OhadRubin opened the pull request:
This pull request fixes https://github.com/facebookresearch/hydra/issues/1898
With the latest version of submitit, tasks_per_node became optional. see [here](https://github.com/facebookincubator/submitit/commit/208198eed20ca77d609bf96b0e89a75bc4ba43eb).
(need to change requirements.txt as well)

Jasha10 reviewed (approved):
Thanks @OhadRubin! Sorry this took so long to review :P

Jasha10 commented on plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py:
```diff
@@ -18,7 +18,7 @@ class BaseQueueConf:
     gpus_per_node: Optional[int] = None
     # number of tasks to spawn on each node
-    tasks_per_node: int = 1
+    tasks_per_node: Optional[int] = None
```
Maybe we should leave the default value as 1 to keep backward compatibility with older versions of the submitit launcher plugin?

OhadRubin committed:
Update config.py

omry committed:
Allow null tasks_per_node in Submitit launcher

Accept null for tasks_per_node while retaining 1 as the default. This allows
Submitit to omit --ntasks-per-node from Slurm submissions.

Add regression coverage, documentation, and a news fragment.

omry commented on plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py:
```diff
@@ -18,7 +18,7 @@ class BaseQueueConf:
     gpus_per_node: Optional[int] = None
     # number of tasks to spawn on each node
-    tasks_per_node: int = 1
+    tasks_per_node: Optional[int] = 1
```
Submitit 1.4.6 already contains the cited optional `ntasks_per_node` implementation ([upstream commit](https://github.com/facebookincubator/submitit/commit/208198eed20ca77d609bf96b0e89a75bc4ba43eb)). Its `AutoExecutor._internal_update_parameters()` also removes `None` values before backend parameter conversion. Therefore the declared minimum version already supports the documented omission behavior; no dependency-bound change is needed.

omry commented on plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py:
```diff
@@ -18,7 +18,7 @@ class BaseQueueConf:
     gpus_per_node: Optional[int] = None
     # number of tasks to spawn on each node
-    tasks_per_node: int = 1
+    tasks_per_node: Optional[int] = 1
```
`submitit.AutoExecutor._internal_update_parameters()` removes all parameters whose value is `None` before delegating to either backend. The local executor therefore never receives `tasks_per_node=None`; the parameter is absent and `LocalExecutor` retains its default of one task. I also verified this directly with Submitit 1.5.4: after `update_parameters(tasks_per_node=None)`, the local backend parameter dictionary remains empty.
